### PR TITLE
feat: upgrade rspack to 0.4.4

### PR DESCRIPTION
**Pull request by ahabhgk**
upgrade Rspack to 0.4.4, https://www.rspack.dev/blog/announcing-0.4.html#rspack-04-major-changes

This contains break change since [`experiments.rspackFuture.disableTransformByDefault`](https://www.rspack.dev/blog/announcing-0.4.html#deprecating-default-transformation) is enabled by default in Rspack 0.4

### CODE REVIEW

File: packages/rspack/src/generators/init/init.ts [hunk 1]
```diff
@@ -40,7 +40,7 @@ export async function rspackInitGenerator(
   }
 
   if (schema.style === 'less') {
-    devDependencies['@rspack/less-loader'] = rspackLessLoaderVersion;
```
Comment (ahabhgk) [inline packages/rspack/src/generators/init/init.ts:43]: @rspack/less-loader is deprecated and should use less-loader instead

**Comment by ahabhgk** [comment]
`FAIL   rspack-e2e  e2e/rspack-e2e/tests/rspack.spec.ts (94.123 s)`, seems e2e of the generated project (`tmp/nx-e2e/proj`) are failed, but it passes when I run `yarn run e2e` under `tmp/nx-e2e/proj`

**Comment by jaysoo** [comment]
There's a weird FreeBSD error for the e2e test. Let me run it locally and check. We'll get this merged. Thanks.

**Comment by jaysoo** [comment]
It looks like `ReactRefresh` is being used in non-dev builds too, so it's causing Cypress test to fail due to React's warning against it in production mode.

File: packages/rspack/src/utils/with-react.ts [hunk 2]
```diff
@@ -15,15 +16,55 @@ export function withReact(opts = {}) {
...
-      builtins: {
-        ...config.builtins,
-        react: {
-          runtime: 'automatic',
-          development: isDev,
-          refresh: isDev,
-        },
+      plugins: [
+        ...(config.plugins || []),
+        new ReactRefreshPlugin(),
```
Comment (jaysoo) [inline packages/rspack/src/utils/with-react.ts:29]: This needs to check if `isDev` then add the plugin, otherwise leave it out.

**Comment by jaysoo** [comment]
It also looks like Nest is failing the e2e test. It cannot compile the decorator in `app.module.ts`.

I see something like this:

```

> nx run app3456491:build:production

PublicPath: /
asset assets/.gitkeep 0 bytes [emitted]
asset main.js 9.73 KiB [emitted] (name: main)
Entrypoint main 9.73 KiB = main.js
runtime modules 9 modules
./app3456491/src/main.ts
external ["@nestjs/core"]
external ["@nestjs/common"]
./app3456491/src/app/app.module.ts
  × Error[javascript]: JavaScript parsing error
   ╭─[app3456491/src/app/app.module.ts:5:1]
 5 │
 6 │ @Module({
   · ┬
   · ╰── Expression expected
 7 │   imports: [],
   ╰────

Rspack 0.4.3 compiled with 1 error in 35 ms (befe8ffbdb5228c52c4b)
```

**Comment by ahabhgk** [comment]
Thanks for your help! There is still a bug on Rspack side which breaks license-webpack-plugin, will ship the fix in 0.4.4, I will update Rspack to 0.4.4 when it is released

**Review by jaysoo — APPROVED** [review]

**Pull request merged**